### ISSUE TITLE
Infrastructure creation failed. Reason: null

### ISSUE THREAD

**Issue by erwan-rouzel** [sequenceiq/cloudbreak-deployer#212]
Hi,

I am currently trying to deploy a Hadoop Cluster using Cloudbreak 1.2.1 with the blueprint hdp-small-default over Openstack and I get the following errors in logs.

I have observed there is Java NullPointerException during the process but not sure to what extent this is relevant to the issue.

Any idea how could I fix that ?

Thanks!

Erwan

================

cloudbreak_1 | 2016-04-15 08:58:28,761 [reactorDispatcher-52] accept:47 DEBUG c.s.c.c.f.Flow2Handler - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] flowcontrol event arrived: key: LAUNCHSTACKRESULT_ERROR, flowid: 2786c842-bf9f-45ab-b33a-837cb01c90fe, payload: LaunchStackResult{status=FAILED, statusReason='', errorDetails=java.lang.NullPointerException, request=CloudStackRequest{, cloudStack=CloudStack{groups=[com.sequenceiq.cloudbreak.cloud.model.Group@1ced9003, com.sequenceiq.cloudbreak.cloud.model.Group@3089c3cb, com.sequenceiq.cloudbreak.cloud.model.Group@24955d52, com.sequenceiq.cloudbreak.cloud.model.Group@4a21c541, com.sequenceiq.cloudbreak.cloud.model.Group@6f1d546c, com.sequenceiq.cloudbreak.cloud.model.Group@6e91b771], network=com.sequenceiq.cloudbreak.cloud.model.Network@20a2946d, security=Security{rules=[SecurityRule{cidr='0.0.0.0/0', ports='[Ljava.lang.String;@11be90d8', protocol='tcp'}]}, image=com.sequenceiq.cloudbreak.cloud.model.Image@76175e01}}, results=null}
cloudbreak_1 | 2016-04-15 08:58:28,803 [reactorDispatcher-52] getResourceDefinition:28 DEBUG c.s.c.s.s.r.d.ResourceDefinitionService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Sending request for OPENSTACK credential resource property definition
cloudbreak_1 | 2016-04-15 08:58:28,804 [reactorDispatcher-53] buildLogContextForReactorHandler:54INFO  c.s.c.l.LogContextAspects - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] A Reactor event handler's 'accept' method has been intercepted: execution(Consumer.accept(..)), MDC logger context is built.
cloudbreak_1 | 2016-04-15 08:58:28,805 [reactorDispatcher-53] accept:30 INFO  c.s.c.c.h.ResourceDefinitionHandler - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Received event: Event{id=null, headers=null, replyTo=null, key=RESOURCEDEFINITIONREQUEST, data=GetResourceDefinition{platform='{platform='OPENSTACK', variant=''}', resource='credential'}}
cloudbreak_1 | 2016-04-15 08:58:28,805 [reactorDispatcher-52] getResourceDefinition:34 INFO  c.s.c.s.s.r.d.ResourceDefinitionService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Resource property definition: ResourceDefinitionResult{definition='{"values":[{"name":"userName","type":"String","encrypted":true},{"name":"password","type":"String","encrypted":true},{"name":"endpoint","type":"String"},{"name":"keystoneVersion","type":"String"}],"selectors":[{"name":"cb-keystone-v2","values":[{"name":"tenantName","type":"String"}]},{"name":"cb-keystone-v3","values":[{"name":"userDomain","type":"String"},{"name":"keystoneAuthScope","type":"String"}]},{"name":"cb-keystone-v3-default-scope","parent":"cb-keystone-v3","values":[]},{"name":"cb-keystone-v3-domain-scope","parent":"cb-keystone-v3","values":[{"name":"domainName","type":"String"}]},{"name":"cb-keystone-v3-project-scope","parent":"cb-keystone-v3","values":[{"name":"projectDomainName","type":"String"},{"name":"projectName","type":"String"}]}]}'}
cloudbreak_1 | 2016-04-15 08:58:28,824 [reactorDispatcher-52] getImage:57 DEBUG c.s.c.s.i.ImageService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Image found! stackId: 3, component: Component{id=2, componentType=IMAGE, name='IMAGE'}
cloudbreak_1 | 2016-04-15 08:58:28,839 [reactorDispatcher-52] handleStackCreationFailure:152 ERROR c.s.c.c.f.s.p.a.StackCreationService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Error during stack creation flow:
cloudbreak_1 | java.lang.NullPointerException: null
cloudbreak_1 |  at com.sequenceiq.cloudbreak.cloud.openstack.heat.OpenStackResourceConnector.launch(OpenStackResourceConnector.java:68) ~[cloud-openstack-1.2.4.jar!/:na]
cloudbreak_1 |  at com.sequenceiq.cloudbreak.cloud.handler.LaunchStackHandler.accept(LaunchStackHandler.java:69) ~[cloud-reactor-1.2.4.jar!/:na]
cloudbreak_1 |  at com.sequenceiq.cloudbreak.cloud.handler.LaunchStackHandler.accept(LaunchStackHandler.java:33) ~[cloud-reactor-1.2.4.jar!/:na]
cloudbreak_1 |  at sun.reflect.GeneratedMethodAccessor190.invoke(Unknown Source) ~[na:na]
cloudbreak_1 |  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_79]
cloudbreak_1 |  at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_79]
cloudbreak_1 |  at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:302) ~[spring-aop-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
cloudbreak_1 |  at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190) ~[spring-aop-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
cloudbreak_1 |  at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157) ~[spring-aop-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
cloudbreak_1 |  at org.springframework.aop.framework.adapter.MethodBeforeAdviceInterceptor.invoke(MethodBeforeAdviceInterceptor.java:52) ~[spring-aop-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
cloudbreak_1 |  at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
cloudbreak_1 |  at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92) ~[spring-aop-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
cloudbreak_1 |  at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
cloudbreak_1 |  at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:208) ~[spring-aop-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
cloudbreak_1 |  at com.sun.proxy.$Proxy199.accept(Unknown Source) ~[na:na]
cloudbreak_1 |  at reactor.bus.EventBus$3.accept(EventBus.java:317) [reactor-bus-2.0.7.RELEASE.jar!/:na]
cloudbreak_1 |  at reactor.bus.EventBus$3.accept(EventBus.java:310) [reactor-bus-2.0.7.RELEASE.jar!/:na]
cloudbreak_1 |  at reactor.bus.routing.ConsumerFilteringRouter.route(ConsumerFilteringRouter.java:72) [reactor-bus-2.0.7.RELEASE.jar!/:na]
cloudbreak_1 |  at reactor.bus.routing.TraceableDelegatingRouter.route(TraceableDelegatingRouter.java:51) [reactor-bus-2.0.7.RELEASE.jar!/:na]
cloudbreak_1 |  at reactor.bus.EventBus.accept(EventBus.java:591) [reactor-bus-2.0.7.RELEASE.jar!/:na]
cloudbreak_1 |  at reactor.bus.EventBus.accept(EventBus.java:63) [reactor-bus-2.0.7.RELEASE.jar!/:na]
cloudbreak_1 |  at reactor.core.dispatch.AbstractLifecycleDispatcher.route(AbstractLifecycleDispatcher.java:160) [reactor-core-2.0.7.RELEASE.jar!/:na]
cloudbreak_1 |  at reactor.core.dispatch.MultiThreadDispatcher$MultiThreadTask.run(MultiThreadDispatcher.java:74) [reactor-core-2.0.7.RELEASE.jar!/:na]
cloudbreak_1 |  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_79]
cloudbreak_1 |  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_79]
cloudbreak_1 |  at java.lang.Thread.run(Thread.java:745) [na:1.7.0_79]
cloudbreak_1 | 2016-04-15 08:58:28,839 [reactorDispatcher-52] fireEventAndLog:203 DEBUG c.s.c.c.f.s.p.a.StackCreationService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] STACK_INFRASTRUCTURE_CREATE_FAILED [STACK_FLOW_STEP].
cloudbreak_1 | 2016-04-15 08:58:28,839 [reactorDispatcher-52] fireCloudbreakEvent:51 INFO  c.s.c.s.e.DefaultCloudbreakEventService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Firing Cloudbreak event: CloudbreakEventData{entityId=3, eventType='UPDATE_IN_PROGRESS', eventMessage='Infrastructure creation failed. Reason: null'}
cloudbreak_1 | 2016-04-15 08:58:28,840 [reactorDispatcher-52] handleFailure:188 DEBUG c.s.c.c.f.s.p.a.StackCreationService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Nothing to do. OnFailureAction DO_NOTHING
cloudbreak_1 | 2016-04-15 08:58:28,840 [reactorDispatcher-54] accept:33 INFO  c.s.c.s.e.CloudbreakEventHandler - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Handling cloudbreak event: Event{id=null, headers=null, replyTo=null, key=CLOUDBREAK_EVENT,data=CloudbreakEventData{entityId=3, eventType='UPDATE_IN_PROGRESS', eventMessage='Infrastructure creation failed. Reason: null'}}
cloudbreak_1 | 2016-04-15 08:58:28,841 [reactorDispatcher-54] accept:36 INFO  c.s.c.s.e.CloudbreakEventHandler - [owner:undefined] [type:CLOUDBREAKEVENTDATA] [id:undefined] [name:undefined] Persisting data: CloudbreakEventData{entityId=3, eventType='UPDATE_IN_PROGRESS', eventMessage='Infrastructure creation failed. Reason: null'}
cloudbreak_1 | 2016-04-15 08:58:28,841 [reactorDispatcher-54] createStackEvent:66 DEBUG c.s.c.s.e.DefaultCloudbreakEventService - [owner:undefined] [type:CLOUDBREAKEVENTDATA] [id:undefined] [name:undefined] Creating stack event from: CloudbreakEventData{entityId=3, eventType='UPDATE_IN_PROGRESS', eventMessage='Infrastructure creation failed. Reason: null'}
cloudbreak_1 | 2016-04-15 08:58:28,883 [reactorDispatcher-54] createStackEvent:71 INFO  c.s.c.s.e.DefaultCloudbreakEventService - [owner:undefined] [type:CLOUDBREAKEVENTDATA] [id:undefined] [name:undefined] Created stack event: CloudbreakEvent{id=4, eventType='UPDATE_IN_PROGRESS', eventTimestamp=Fri Apr 15 08:58:28 UTC 2016, eventMessage='Infrastructure creation failed. Reason: null',owner='a3beda67-ebbb-4bba-b16c-ca7d73e26e88', account='seq1234567', cloud='OPENSTACK', region='local', blueprintName='hdp-small-default', instanceGroup='null', blueprintId=1', stackId=3', clusterId=2', clusterName=demo-cluster-2}
cloudbreak_1 | 2016-04-15 08:58:28,884 [reactorDispatcher-54] accept:38 INFO  c.s.c.s.e.CloudbreakEventHandler - [owner:undefined] [type:CLOUDBREAKEVENTDATA] [id:undefined] [name:undefined] Sending notification with data: CloudbreakEvent{id=4, eventType='UPDATE_IN_PROGRESS', eventTimestamp=Fri Apr 15 08:58:28 UTC 2016, eventMessage='Infrastructure creation failed. Reason: null', owner='a3beda67-ebbb-4bba-b16c-ca7d73e26e88', account='seq1234567', cloud='OPENSTACK', region='local',blueprintName='hdp-small-default', instanceGroup='null', blueprintId=1', stackId=3', clusterId=2', clusterName=demo-cluster-2}
cloudbreak_1 | 2016-04-15 08:58:28,899 [reactorDispatcher-52] fireEventAndLog:203 DEBUG c.s.c.c.f.s.p.a.StackCreationService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] STACK_INFRASTRUCTURE_CREATE_FAILED [STACK_FLOW_STEP].
cloudbreak_1 | 2016-04-15 08:58:28,899 [reactorDispatcher-52] fireCloudbreakEvent:51 INFO  c.s.c.s.e.DefaultCloudbreakEventService - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Firing Cloudbreak event: CloudbreakEventData{entityId=3, eventType='CREATE_FAILED', eventMessage='Infrastructure creation failed. Reason: null'}
cloudbreak_1 | 2016-04-15 08:58:28,899 [reactorDispatcher-52] sendEvent:56 INFO  c.s.c.c.f.AbstractAction - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2]Triggering event: com.sequenceiq.cloudbreak.core.flow2.stack.FlowFailureEvent@54075261
cloudbreak_1 | 2016-04-15 08:58:28,900 [reactorDispatcher-55] accept:33 INFO  c.s.c.s.e.CloudbreakEventHandler - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] Handling cloudbreak event: Event{id=null, headers=null, replyTo=null, key=CLOUDBREAK_EVENT,data=CloudbreakEventData{entityId=3, eventType='CREATE_FAILED', eventMessage='Infrastructure creation failed. Reason: null'}}
cloudbreak_1 | 2016-04-15 08:58:28,900 [reactorDispatcher-55] accept:36 INFO  c.s.c.s.e.CloudbreakEventHandler - [owner:undefined] [type:CLOUDBREAKEVENTDATA] [id:undefined] [name:undefined] Persisting data: CloudbreakEventData{entityId=3, eventType='CREATE_FAILED', eventMessage='Infrastructure creation failed. Reason: null'}
cloudbreak_1 | 2016-04-15 08:58:28,900 [reactorDispatcher-55] createStackEvent:66 DEBUG c.s.c.s.e.DefaultCloudbreakEventService - [owner:undefined] [type:CLOUDBREAKEVENTDATA] [id:undefined] [name:undefined] Creating stack event from: CloudbreakEventData{entityId=3, eventType='CREATE_FAILED', eventMessage='Infrastructure creation failed. Reason: null'}
cloudbreak_1 | 2016-04-15 08:58:28,900 [reactorDispatcher-52] stateChanged:82 INFO  c.s.c.c.f.s.p.StackCreationFlowConfig - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] StackCreationFlow changed from ObjectState [getIds()=[START_PROVISIONING_STATE],getClass()=class org.springframework.statemachine.state.ObjectState, hashCode()=689975318, toString()=AbstractState [id=START_PROVISIONING_STATE, pseudoState=null, deferred=null, entryActions=[com.sequenceiq.cloudbreak.core.flow2.stack.provision.action.StartProvisioningAction@350972], exitActions=null, regions=[], submachine=null]] to ObjectState [getIds()=[STACK_CREATION_FAILED_STATE], getClass()=class org.springframework.statemachine.state.ObjectState, hashCode()=1600144645, toString()=AbstractState [id=STACK_CREATION_FAILED_STATE, pseudoState=null, deferred=null, entryActions=[com.sequenceiq.cloudbreak.core.flow2.stack.provision.action.StackCreationFailureAction@7ea50c74], exitActions=null, regions=[], submachine=null]]
cloudbreak_1 | 2016-04-15 08:58:28,901 [reactorDispatcher-56] accept:47 DEBUG c.s.c.c.f.Flow2Handler - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] flowcontrol event arrived: key: FAILHANDLED, flowid: 2786c842-bf9f-45ab-b33a-837cb01c90fe, payload:com.sequenceiq.cloudbreak.core.flow2.stack.FlowFailureEvent@54075261
cloudbreak_1 | 2016-04-15 08:58:28,901 [reactorDispatcher-56] sendEvent:56 INFO  c.s.c.c.f.AbstractAction - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2]Triggering event: com.sequenceiq.cloudbreak.core.flow2.stack.FlowFailureEvent@54075261
cloudbreak_1 | 2016-04-15 08:58:28,902 [reactorDispatcher-56] stateChanged:82 INFO  c.s.c.c.f.s.p.StackCreationFlowConfig - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] StackCreationFlow changed from ObjectState [getIds()=[STACK_CREATION_FAILED_STATE], getClass()=class org.springframework.statemachine.state.ObjectState, hashCode()=1600144645, toString()=AbstractState [id=STACK_CREATION_FAILED_STATE, pseudoState=null, deferred=null, entryActions=[com.sequenceiq.cloudbreak.core.flow2.stack.provision.action.StackCreationFailureAction@7ea50c74], exitActions=null, regions=[], submachine=null]] to ObjectState [getIds()=[FINAL_STATE], getClass()=class org.springframework.statemachine.state.ObjectState, hashCode()=3095445, toString()=AbstractState [id=FINAL_STATE, pseudoState=org.springframework.statemachine.state.DefaultPseudoState@1cf0b34b, deferred=null, entryActions=[com.sequenceiq.cloudbreak.core.flow2.FlowFinalizeAction@3a04c35d], exitActions=null, regions=[], submachine=null]]
cloudbreak_1 | 2016-04-15 08:58:28,902 [reactorDispatcher-56] stop:127 INFO  o.s.s.s.LifecycleObjectSupport - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] stopped org.springframework.statemachine.support.DefaultStateMachineExecutor@1eb53f7b
cloudbreak_1 | 2016-04-15 08:58:28,902 [reactorDispatcher-57] accept:56 DEBUG c.s.c.c.f.Flow2Handler - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] flowfinalizing arrived: id: 2786c842-bf9f-45ab-b33a-837cb01c90fe
cloudbreak_1 | 2016-04-15 08:58:28,902 [reactorDispatcher-56] stop:127 INFO  o.s.s.s.LifecycleObjectSupport - [owner:a3beda67-ebbb-4bba-b16c-ca7d73e26e88] [type:STACK] [id:3] [name:demo-cluster-2] stopped PROVISIONING_FINISHED_STATE START_PROVISIONING_STATE SETUP_STATE IMAGE_CHECK_STATE IMAGESETUP_STATE STACK_CREATION_FAILED_STATE TLS_SETUP_STATE FINAL_STATE INIT_STATE COLLECTMETADATA_STATE  /  / id=b08412b4-ab37-40d5-a19b-d53f2e923bdf
cloudbreak_1 | 2016-04-15 08:58:28,930 [reactorDispatcher-55] createStackEvent:71 INFO  c.s.c.s.e.DefaultCloudbreakEventService - [owner:undefined] [type:CLOUDBREAKEVENTDATA] [id:undefined] [name:undefined] Created stack event: CloudbreakEvent{id=5, eventType='CREATE_FAILED', eventTimestamp=Fri Apr 15 08:58:28 UTC 2016, eventMessage='Infrastructure creation failed. Reason: null', owner='a3beda67-ebbb-4bba-b16c-ca7d73e26e88', account='seq1234567', cloud='OPENSTACK', region='local',blueprintName='hdp-small-default', instanceGroup='null', blueprintId=1', stackId=3', clusterId=2', clusterName=demo-cluster-2}
cloudbreak_1 | 2016-04-15 08:58:28,930 [reactorDispatcher-55] accept:38 INFO  c.s.c.s.e.CloudbreakEventHandler - [owner:undefined] [type:CLOUDBREAKEVENTDATA] [id:undefined] [name:undefined] Sending notification with data: CloudbreakEvent{id=5, eventType='CREATE_FAILED', eventTimestamp=Fri Apr 15 08:58:28 UTC 2016, eventMessage='Infrastructure creation failed. Reason: null', owner='a3beda67-ebbb-4bba-b16c-ca7d73e26e88', account='seq1234567', cloud='OPENSTACK', region='local', blueprintName='hdp-small-default', instanceGroup='null', blueprintId=1', stackId=3', clusterId=2', clusterName=demo-cluster-2}

**Comment by doktoric** [comment]
Which version of Openstack are you using?

**Comment by erwan-rouzel** [comment]
This is Openstack Juno.

Also one information in case it can help : we have a proxy and need to do some special configuration in Openstack in case the deployment needs to do connection to internet. Is it the case or everything happens internal ?

**Comment by doktoric** [comment]
deployment needs connection with you openstack cluster. The internet needs just for the deployed nodes because ambari installing the packages from the central yum repositories.

Could you please send us the whole log ? my mail is: rdoktorics@hortonworks.com

Br,
R

**Comment by erwan-rouzel** [comment]
Ok, I have just sent you the complete log to your mail. Hope this will help.

**Comment by doktoric** [comment]
If you are using Openstack there are two option to provision a VM.
First when you provision a VM with heat templates. As I see your Openstack not supporting Heat templates or it is just simply not enabled. 
If you dont want to enabled this feature in Openstack, then use the native implementation option which is available if you clicking on the 'Show Advanced Options' button on the cluster creation form.
Br,
R

**Comment by erwan-rouzel** [comment]
Thank you for your answer. You are right that Heat templates are not enabled on this Openstack so this was the reason.

I used the "native" option as suggested which sorted out the issue. I faced few other issues due to setup of network and instances definition (size of disk had to be 100GB) which was not due to Cloudbreak but only configuration on the side of Openstack and finally it all worked fine and could deploy the first cluster successfully!
